### The dataset that I used for this project was the Titantic Data set. The questions that I posed were to understand what three factors had the highest impact on the chance of survival for the passengers. I focused my analysis on the gender of the passenger, the class of ticket that was purchased, as well as the age of the passengers. I did find in the dataset that there were passengers whose age were not known, so I performed my age analysis on passengers whose ages were known. My steps to analyzing the data are described below. My analysis can be found below.

In [2]:
#Importing the data into the notebook

import pandas as pd

df = pd.read_csv("titanic_data.csv")

print df.head()

   PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex  Age  SibSp  \
0                            Braund, Mr. Owen Harris    male   22      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female   38      1   
2                             Heikkinen, Miss. Laina  female   26      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female   35      1   
4                           Allen, Mr. William Henry    male   35      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S  


### This part of the Analysis is used to load the dataset into the Python notebook

In [3]:
#Amount of passengers within the dataset

len(df)

891

### The dataset contains one row of information per passenger. The length of the dataset was used in order to count the amount of passengers within the dataset. The length of the dataset is 891, meaning there are 891 passengers within the dataset

In [4]:
#Cleansing Data to remove unnecessary columns not needed for Analysis

df = df.drop(['Name', 'Ticket', 'Fare', 'Embarked', 'Cabin'], axis = 1)
df.head(3)

PassengerId  Survived  Pclass     Sex  Age  SibSp  Parch
0            1         0       3    male   22      1      0
1            2         1       1  female   38      1      0
2            3         1       3  female   26      0      0

### There were several columns within the dataset that were not imparative to analyze in regards to factors pertaining to the survivors. The passengers' names, price of ticket, embarcation point, as well as their cabin did not have a direct impact on whether the passengers survived or not. The ticket price correlated to the class of the ticket, however the amount that was paid for the ticket did not have an affect on the survival rate.

In [5]:
#All Passengers broken down by Gender

total_df = df[['PassengerId', 'Sex']]
total_df.groupby(['Sex']).count()

PassengerId
Sex                
female          314
male            577

### The dataset was then broken down for all passengers, whether they survived or not, by gender. There were more males travelling than females in this dataset by 30%.

In [6]:
#Amount of Survived Passengers broken down by Gender

total_survivors = df[(df['Survived'] == 1)]
survivors = total_survivors[['Survived', 'Sex']]
survivors.groupby(['Sex']).count()

Survived
Sex             
female       233
male         109

### The dataset was then broken down by whether the passenger survived if there was a '1' in the Survived column within their row. I broke down the gender of survivors and found that out of the 342 survivors within this dataset, 68% were females. This can be considered a factor of survival because it represents that for every male who survived, 3 females survived.

In [7]:
#Survived Passengers broken down by Gender and Class of Ticket

total_survivors = df[(df['Survived'] == 1)]
survived = total_survivors[['Survived', 'Pclass', 'Sex']]
survived.groupby(['Sex', 'Pclass']).count()


Survived
Sex    Pclass          
female 1             91
       2             70
       3             72
male   1             45
       2             17
       3             47

### I then broke the dataset down by gender and by their class of ticket. For females, there was the same amount of survivors in either second or third class, whereas females who purchased first class tickets had a higher survival rate. For males, they had clost to the same chance of survival if they purchased either a first or third class ticket. The class of ticket is considered a factor of survival because the data shows that there more survivors for either females or males in certain classes of tickets purchased.

In [8]:
#Survived Females broken down by Class of Ticket

total_females = df[(df['Sex'] == 'female') & (df['Survived'] == 1)]
survived_females = total_females[['Survived', 'Pclass']]
survived_females.groupby(['Pclass']).count()


Survived
Pclass          
1             91
2             70
3             72

### The above dataset shows the survived females broken down by class separately.

In [9]:
#Survived Males broken down by Class of Ticket

total_males = df[(df['Sex'] == 'male') & (df['Survived'] == 1)]
survived_males = total_males[['Survived', 'Pclass']]
survived_males.groupby(['Pclass']).count()


Survived
Pclass          
1             45
2             17
3             47

### The above dataset shows the survived males broken down by class separately.

In [10]:
#Average Age of Survived Females 

total_females_description = df[(df['Sex'] == 'female') & (df['Survived'] == 1)]
decimals = total_females_description.mean()['Age']
decimals.round()


29.0

### The average age then was taken for survived females to understand if age was a factor in survival. The average age of surviving females was 29 years old.

In [11]:
#Average Age of Survived Males

total_males_description = df[(df['Sex'] == 'male') & (df['Survived'] == 1)]
decimals = total_males_description.mean()['Age']
decimals.round()

27.0

### The average age was then taken for survived males and was found that the average age of surviving males was 27 years old.

In [12]:
import numpy as np
import matplotlib.pyplot as plt

#Survived Males Ages Histograms by Class of Ticket

a = df[(df['Sex'] == 'male') & (df['Survived'] == 1)&(pd.notnull(df['Age']))]['Age']
b = df[(df['Sex'] == 'male') & (df['Survived'] == 1)&(pd.notnull(df['Age']))&(df['Pclass'] == 1)]['Age']
c = df[(df['Sex'] == 'male') & (df['Survived'] == 1)&(pd.notnull(df['Age']))&(df['Pclass'] == 2)]['Age']
d = df[(df['Sex'] == 'male') & (df['Survived'] == 1)&(pd.notnull(df['Age']))&(df['Pclass'] == 3)]['Age']




fig1 = plt.figure()
ax1 = fig1.add_subplot(1, 1, 1)
n, bins, patches = ax1.hist(a, color = '#286DA8')
ax1.set_ylim(0, 40)
ax1.set_xlabel('Age of Males Survived', fontsize=14)
ax1.set_ylabel('Survived Males Count', fontsize=14)
ax1.set_title('Ages of Survived Males for all Classes', fontsize=16)
ax1.spines["top"].set_visible(False)  
ax1.spines["right"].set_visible(False)
ax1.get_xaxis().tick_bottom()  
ax1.get_yaxis().tick_left()

  

fig2 = plt.figure()
ax2 = fig2.add_subplot(1, 1, 1)
n, bins, patches = ax2.hist(b, color = '#CD5360')
ax2.set_ylim(0, 15)
ax2.set_xlabel('Age of Males Survived', fontsize=14)
ax2.set_ylabel('Survived Males Count', fontsize=14)
ax2.set_title('Ages of Survived Males in First Class', fontsize=16)
ax2.spines["top"].set_visible(False)  
ax2.spines["right"].set_visible(False)
ax2.get_xaxis().tick_bottom()  
ax2.get_yaxis().tick_left() 

fig3 = plt.figure()
ax3 = fig3.add_subplot(1, 1, 1)
ax3.set_ylim(0, 15)
n, bins, patches = ax3.hist(c, color = '#B37D4E')
ax3.set_xlabel('Age of Males Survived', fontsize=14)
ax3.set_ylabel('Survived Males Count', fontsize=14)
ax3.set_title('Ages of Survived Males in Second Class', fontsize=16)
ax3.spines["top"].set_visible(False)  
ax3.spines["right"].set_visible(False)
ax3.get_xaxis().tick_bottom()  
ax3.get_yaxis().tick_left() 

fig4 = plt.figure()
ax4 = fig4.add_subplot(1, 1, 1)
n, bins, patches = ax4.hist(d, color = '#438496')
ax4.set_ylim(0, 15)
ax4.set_xlabel('Age of Males Survived', fontsize=14)
ax4.set_ylabel('Survived Males Count', fontsize=14)
ax4.set_title('Ages of Survived Males in Third Class', fontsize=16)
ax4.spines["top"].set_visible(False)  
ax4.spines["right"].set_visible(False)
ax4.get_xaxis().tick_bottom()  
ax4.get_yaxis().tick_left() 

### Histograms of the survived passengers' ages are a great visualization to use to analyze the survivors' ages compared by their class of ticket. The first histogram represents all ages of survived males, including all three types of ticket classes. There is a spike within the first histogram within the age group of 25 - 30, which correlates to the average age of male survivors, regardless of their class of ticket. The second histogram then represents survived males in the first class and the most frequent age is between 35 - 40. The third histogram shows survived males in the second class representing a spike within the 0 - 5 year age group. The last histogram of the survived males in the third class show that males ranging between 22 - 27 years old had a higher chance of survival.

In [15]:
#Survived Females Ages Histograms by Class of Ticket

a2 = df[(df['Sex'] == 'female') & (df['Survived'] == 1)&(pd.notnull(df['Age']))]['Age']
b2 = df[(df['Sex'] == 'female') & (df['Survived'] == 1)&(pd.notnull(df['Age']))&(df['Pclass'] == 1)]['Age']
c2 = df[(df['Sex'] == 'female') & (df['Survived'] == 1)&(pd.notnull(df['Age']))&(df['Pclass'] == 2)]['Age']
d2 = df[(df['Sex'] == 'female') & (df['Survived'] == 1)&(pd.notnull(df['Age']))&(df['Pclass'] == 3)]['Age']




fig1 = plt.figure()
fax1 = fig1.add_subplot(1, 1, 1)
n, bins, patches = fax1.hist(a2, color = '#286DA8')
fax1.set_ylim(0, 40)
fax1.set_xlabel('Age of Females Survived', fontsize=14)
fax1.set_ylabel('Survived Females Count', fontsize=14)
fax1.set_title('Ages of Survived Females for all Classes', fontsize=16)
fax1.spines["top"].set_visible(False)  
fax1.spines["right"].set_visible(False)
fax1.get_xaxis().tick_bottom()  
fax1.get_yaxis().tick_left()

  

fig2 = plt.figure()
fax2 = fig2.add_subplot(1, 1, 1)
n, bins, patches = fax2.hist(b2, color = '#CD5360')
fax2.set_ylim(0, 15)
fax2.set_xlabel('Age of Females Survived', fontsize=14)
fax2.set_ylabel('Survived Females Count', fontsize=14)
fax2.set_title('Ages of Survived Females in First Class', fontsize=16)
fax2.spines["top"].set_visible(False)  
fax2.spines["right"].set_visible(False)
fax2.get_xaxis().tick_bottom()  
fax2.get_yaxis().tick_left() 

fig3 = plt.figure()
fax3 = fig3.add_subplot(1, 1, 1)
fax3.set_ylim(0, 15)
n, bins, patches = fax3.hist(c2, color = '#B37D4E')
fax3.set_xlabel('Age of Females Survived', fontsize=14)
fax3.set_ylabel('Survived Females Count', fontsize=14)
fax3.set_title('Ages of Survived Females in Second Class', fontsize=16)
fax3.spines["top"].set_visible(False)  
fax3.spines["right"].set_visible(False)
fax3.get_xaxis().tick_bottom()  
fax3.get_yaxis().tick_left() 

fig4 = plt.figure()
fax4 = fig4.add_subplot(1, 1, 1)
n, bins, patches = fax4.hist(d2, color = '#438496')
fax4.set_ylim(0, 15)
fax4.set_xlabel('Age of Females Survived', fontsize=14)
fax4.set_ylabel('Survived Females Count', fontsize=14)
fax4.set_title('Ages of Survived Females in Third Class', fontsize=16)
fax4.spines["top"].set_visible(False)  
fax4.spines["right"].set_visible(False)
fax4.get_xaxis().tick_bottom()  
fax4.get_yaxis().tick_left() 

### The second set of histograms represent survived females' ages grouped by their class of ticket. The first histogram of survived females reflect females between the ages of 15 to 35 had the highest rate of survival. The second histogram represents survived females who purchased a first class ticket whose age ranged from 29 to 39. The third histogram shows survived females in the second class ranging between the age of 23 - 29 had the highest rate of survival. The last histogram of survived females who purchased third class tickets had the highest rate of survival between the ages of 15 - 20.

## Conclusion
### This dataset representing a portion of passengers who travelled on the Titantic can be interpretted that females who purchased a first class ticket with an average age of 29 had the highest rate of survival. Males had the highest rate of survival if they purchased either a first class or third class ticket whose age averaged 27. There cannot be assumptions made within this dataset that more females and children survived if prior knowledge was not known about Titantic survivors. 